src/main/java/io/github/thebusybiscuit/slimefun4/implementation/operations/FuelOperation.java

```diff
@@ -0,0 +1,58 @@
+package io.github.thebusybiscuit.slimefun4.implementation.operations;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.slimefun4.core.machines.MachineOperation;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+
+public class FuelOperation implements MachineOperation {
+
+    private final ItemStack ingredient;
+    private final ItemStack result;
+
+    private final int totalTicks;
+    private int currentTicks = 0;
+
+    public FuelOperation(@Nonnull MachineFuel recipe) {
+        this(recipe.getInput(), recipe.getOutput(), recipe.getTicks());
+    }
+
+    public FuelOperation(@Nonnull ItemStack ingredient, @Nullable ItemStack result, int totalTicks) {
+        Validate.notNull(ingredient, "The Ingredient cannot be null");
+        Validate.isTrue(totalTicks > 0, "The amount of total ticks must be a positive integer");
+
+        this.ingredient = ingredient;
+        this.result = result;
+        this.totalTicks = totalTicks;
+    }
+
+    @Override
+    public void addProgress(int num) {
```

Comment:
Maybe a check to make sure its a positive number?
Or start using a uint I suppose